api_app/analyzers_manager/observable_analyzers/ailtyposquatting.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import math
+
+from ail_typo_squatting import addDash, omission, subdomain
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+
+logger = logging.getLogger(__name__)
+
+
+class AilTypoSquatting(classes.ObservableAnalyzer):
+    """
+    wrapper for https://github.com/typosquatter/ail-typo-squatting
+    """
+
+    omission: bool = False
+    subdomain: bool = False
+    addDash: bool = False
+    runall: bool = True
+
+    def update(self) -> bool:
+        pass
+
+    def run(self):
+        resultList = []
+        response = {}
+        limit = math.inf
+        if self.runall or self.omission:
+            response["omission"] = omission(
+                domain=self.observable_name,
+                resultList=resultList,
+                verbose=False,
+                limit=limit,
+                givevariations=False,
+                keeporiginal=False,
+            )
+        if self.runall or self.subdomain:
+            response["subdomain"] = subdomain(
+                domain=self.observable_name,
+                resultList=resultList,
+                verbose=False,
+                limit=limit,
+                givevariations=False,
+                keeporiginal=False,
+            )
+        if self.runall or self.addDash:
+            response["addDash"] = addDash(
+                domain=self.observable_name,
+                resultList=resultList,
+                verbose=False,
+                limit=limit,
+                givevariations=False,
+                keeporiginal=False,
+            )
+        else:
+            raise AnalyzerRunException("No algo selected to run")
```

Comment:
considering that there are too many algorithms and that the analyzer would be too much complicated by integrating all of them, I think you can just integrate the[ "runAll" function without any params](https://github.com/typosquatter/ail-typo-squatting?tab=readme-ov-file#to-run-all-algorithms) and that's all for algorithms.

However, an interesting param to add is [the "DNS resolving feature"](https://github.com/typosquatter/ail-typo-squatting?tab=readme-ov-file#dns-output). This parameter can work only if the TLP has been set as CLEAR (you should check it directly inside the analyzer itself)

Reply:
there are only three algos are you sure?

Reply:
https://github.com/typosquatter/ail-typo-squatting?tab=readme-ov-file#list-of-algorithms-used

Reply:
holy cow :crying_cat_face: ! let's do `runAll` only then